app/src/main/java/org/kinecosystem/kinit/util/BindingUtils.java

```diff
@@ -127,4 +131,33 @@ public static void setSelectionListener(QuizAnswerView view, QuizAnswerView.OnSe
             view.setListener(listener);
         }
     }
+
+    @BindingAdapter("onType")
+    public static void setInputListener(EditText view, TextWatcher listener) {
+        if (listener != null) {
+            view.addTextChangedListener(listener);
+        }
+    }
+
+    @BindingAdapter("onSpinnerItemSelected")
+    public static void setSpinnerItemListener(Spinner view, AdapterView.OnItemSelectedListener listener) {
+        if (listener != null) {
+            view.setOnItemSelectedListener(listener);
+        }
+    }
+
+    @BindingAdapter("visibilityOn")
+    public static void setVisibilityOn(EditText view, boolean visible) {
```

Comment:
this both sets the visibility on/off and also opens/closes keyboard - should be separated to 2 methods